docs/source/API/core/Macros.rst

```diff
@@ -254,3 +256,8 @@ Architectures
 | ``KOKKOS_ARCH_VEGA90A``           | Optimize for AMD GPU MI200 series GFX90A (GPU)                                    |
 +-----------------------------------+-----------------------------------------------------------------------------------+
 
+*) ``KOKKOS_ARCH_INTEL_GEN`` enables Just-In-Time compilation for Intel GPUs whereas all the other flags for Intel compilers
+request Ahead-Of-Time compilation. Just-In-Time compilation means that the compiler is invoked again when the binaries created
+are actually executed and only at that point the architecture to compile for is determined. On the other hand, Ahead-Of-Time
+compilation describes the standard model where the compiler is only invoked once to create the binary and the architecture to
+compile for is determined before the program is run.
```

Comment:
I just realized you commented by the macros.  It is actually more useful by the CMake options